src/Task/Stylelint.php

```diff
@@ -0,0 +1,140 @@
+        $resolver->addAllowedTypes('report_invalid_scope_disables', ['null', 'bool']);
+        $resolver->addAllowedTypes('report_descriptionless_disables', ['null', 'bool']);
+        $resolver->addAllowedTypes('max_warnings', ['null', 'integer']);
+        $resolver->addAllowedTypes('output_file', ['null', 'string']);
+
+        return $resolver;
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+
+        $files = $context
+            ->getFiles()
+            ->paths($config['whitelist_patterns'] ?? [])
+            ->extensions($config['triggered_by']);
+
+        if (0 === \count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = isset($config['bin'])
+            ? ProcessArgumentsCollection::forExecutable($config['bin'])
+            : $this->processBuilder->createArgumentsForCommand('stylelint');
+
+        $arguments->addOptionalArgument('--config=%s', $config['config']);
+        $arguments->addOptionalArgument('--config-basedir=%s', $config['config_basedir']);
+        $arguments->addOptionalArgument('--ignore-path=%s', $config['ignore_path']);
+        $arguments->addOptionalArgument('--ignore-pattern=%s', $config['ignore_pattern']);
+        $arguments->addOptionalArgument('--syntax=%s', $config['syntax']);
+        $arguments->addOptionalArgument('--custom-syntax=%s', $config['custom_syntax']);
+        $arguments->addOptionalArgument('--ignore-disables', $config['ignore_disables']);
+        $arguments->addOptionalArgument('--disable-default-ignores', $config['disable_default_ignores']);
+        $arguments->addOptionalArgument('--cache', $config['cache']);
+        $arguments->addOptionalArgument('--cache-location=%s', $config['cache_location']);
+        $arguments->addOptionalArgument('--formatter=%s', $config['formatter']);
+        $arguments->addOptionalArgument('--custom-formatter=%s', $config['custom_formatter']);
+        $arguments->addOptionalArgument('--quiet', $config['quiet']);
+        $arguments->addOptionalBooleanArgument('--%s', $config['color'], 'color', 'no-color');
+        $arguments->addOptionalArgument('--report-needless-disables', $config['report_needless_disables']);
+        $arguments->addOptionalArgument('--report-invalid-scope-disables', $config['report_invalid_scope_disables']);
+        $arguments->addOptionalArgument(
+            '--report-descriptionless-disables',
+            $config['report_descriptionless_disables']
+        );
+        $arguments->addOptionalIntegerArgument('--max-warnings=%d', $config['max_warnings']);
+        $arguments->addOptionalArgument('--output-file=%s', $config['output_file']);
+        $arguments->addFiles($files);
```

Comment:
Should files always be added? Or should we only add it during pre-commit?
The full list can be quite big, making it crash on windows-like systems which have a smaller max CLI input length

Reply:
In that case, it seems like a better solution to use stdin instead?

Example:
https://github.com/phpro/grumphp/blob/master/src/Task/PhpLint.php#L63-L70

You'll also need to provide an stdin for the fix process.

Reply:
I tried to run `echo "path_to_an_scss_file.scss" | node_modules/.bin/stylelint` and I got a `CssSyntaxError`, so it seems that Stylelint support direct SCSS code as input with stdin, not a list of file like in the example (if I understood correctly).
I'm completely fine to add the list of files only in the precommit context, as it's there where it's the most useful, and for the run context tools like stylelint usually have a config file
is it ok for you?

Reply:
@veewee updated with adding files only on precommit